### PR TITLE
Update 4FGL catalog default  to DR4 

### PR DESCRIPTION
**Pull request by QRemy**
Update 4FGL catalog default to DR4 and load the correct extended sources file based on the catalog version.
Requires https://github.com/gammapy/gammapy-data/pull/42

### CODE REVIEW

**Review by registerrier — COMMENTED** [review]
Thanks @QRemy . Is it possible to add tests using different versions of the release?

File: gammapy/catalog/fermi.py [hunk 1]
```diff
@@ -1255,18 +1259,19 @@ class SourceCatalog4FGL(SourceCatalog):
...
     One source is represented by `~gammapy.catalog.SourceCatalogObject4FGL`.
     """
 
     tag = "4fgl"
-    description = "LAT 8-year point source catalog"
+    description = "LAT 14-year point source catalog"
     source_object_class = SourceCatalogObject4FGL
 
-    def __init__(self, filename="$GAMMAPY_DATA/catalogs/fermi/gll_psc_v28.fit.gz"):
+    def __init__(self, filename="$GAMMAPY_DATA/catalogs/fermi/gll_psc_v32.fit"):
```
[the remark this answers is not in the document]
Comment (QRemy) [reply gammapy/catalog/fermi.py:1274]: done, also I updated the tests based DR2&3 to DR4 so we can delete the catalog files from DR2 and DR3 from gammapy-data

[the remark this answers is not in the document]
Comment (bkhelifi) [reply gammapy/catalog/fermi.py:1274]: So, the code should be updated after you change, no?

**Review by bkhelifi — APPROVED** [review]
Thanks a lot @QRemy

**Comment by registerrier** [comment]
Regarding the backports: for former versions we use tagged versions of gammapy-data. The changes you introduced should therefore not appear in gammapy-data used by v1.0.x and v1.1.x.

File: gammapy/catalog/fermi.py (hunk 1 above)
Comment (registerrier) [inline gammapy/catalog/fermi.py:1274]:
```suggestion
    def __init__(self, filename="$GAMMAPY_DATA/catalogs/fermi/gll_psc_v32.fit.gz"):
```

**Review by registerrier — APPROVED** [review]
Thanks @QRemy. Looks good!

I have removed the backports here.

**Pull request merged**